F1H43C_EEJYN9/Battleship/Core/PreferenceManager.cs

```diff
@@ -0,0 +1,156 @@
+
+    public PreferenceManager(string preferencesDirectory = "preferences")
+    {
+        _preferencesDirectory = preferencesDirectory;
+        Directory.CreateDirectory(_preferencesDirectory);
+        
+        _availableColors = Enum.GetValues(typeof(ConsoleColor))
+            .Cast<ConsoleColor>()
+            .ToList();
+
+        _defaultPreferences = new GamePreferences();
+    }
+
+    public bool SavePreferences(string username, GamePreferences preferences)
+    {
+        if (string.IsNullOrWhiteSpace(username))
+            throw new ArgumentException("Username cannot be empty", nameof(username));
+
+        ValidateAndApplyDefaults(preferences);
+
+        try
+        {
+            string filePath = GetPreferenceFilePath(username);
+            string jsonString = JsonSerializer.Serialize(preferences, new JsonSerializerOptions
+            {
+                WriteIndented = true
+            });
+
+            _ = Directory.CreateDirectory(Path.GetDirectoryName(filePath));
+            File.WriteAllText(filePath, jsonString);
+            return true;
+        }
+        catch (IOException ioe)
+        {
+            Console.WriteLine($"IO Error saving preferences: {ioe.Message}");
+            return false;
+        }
+        catch (JsonException je)
+        {
+            Console.WriteLine($"Serialization Error saving preferences: {je.Message}");
+            return false;
+        }
+        catch (Exception e)
+        {
+            Console.WriteLine($"Unexpected Error saving preferences: {e.Message}");
+            return false;
+        }
+    }
+
+    public GamePreferences LoadPreferences(string username)
+    {
+        if (string.IsNullOrWhiteSpace(username))
+            throw new ArgumentException("Username cannot be empty", nameof(username));
+
+        try
+        {
+            string filePath = GetPreferenceFilePath(username);
+            if (!File.Exists(filePath))
+                return new GamePreferences();
+
+            string jsonString = File.ReadAllText(filePath);
+            var preferences = JsonSerializer.Deserialize<GamePreferences>(jsonString);
+            ValidateAndApplyDefaults(preferences);
+            return preferences;
+        }
+        catch (IOException ioe)
+        {
+            Console.WriteLine($"Error loading preferences: {ioe.Message}");
+            return new GamePreferences();
+        }
+        catch (JsonException je)
+        {
+            Console.WriteLine($"Error parsing preferences: {je.Message}");
+            return new GamePreferences();
+        }
+        catch (Exception e)
+        {
+            Console.WriteLine($"Unexpected Error saving preferences: {e.Message}");
+       return new GamePreferences();
+        }
+    }
+
+    public List<ConsoleColor> GetAvailableColors()
+    {
+        return _availableColors.ToList();
+    }
+
+    public void ValidateAndApplyDefaults(GamePreferences preferences)
+    {
```

Comment:
Mivel ez JSON akár egy JSON Schema alapú validációval is megoldható lenne. Egyetlen baj, hogy a .NET beépített megoldást még nem tartalmaz JSON Schema olvasására.